src/utils/reducers/themeReducer.js

```diff
@@ -0,0 +1,17 @@
+import themes from '../../assets/themes';
+
+export const initialTheme = {
+  activeTheme: 'light',
+  theme: themes.light,
+};
+
+export function themeReducer(state, action) {
+  switch (action.type) {
+    case 'toggleTheme': {
+      const newThemeKey = state.activeTheme === 'light' ? 'dark' : 'light';
+      return { activeTheme: newThemeKey, theme: themes[newThemeKey] };
+    }
+    default:
+      throw new Error();
+  }
+}
```

Comment:
Would you mind adding a test to assert that the reducer actually toggles the state of `activeTheme`?

Reply:
Also, one of the acceptance criteria for this challenge is to save the `searchTerm` into the global context and retrieve it from there, you're missing this.